### PR TITLE
refactor(llmq): misc refactors

### PR DESCRIPTION
**Pull request by PastaPastaPasta**

### CODE REVIEW

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: src/llmq/commitment.h [hunk 1]
```diff
@@ -100,8 +100,8 @@ class CFinalCommitment
     void ToJson(UniValue& obj) const
     {
         obj.setObject();
-        obj.pushKV("version", (int)nVersion);
-        obj.pushKV("llmqType", (int)llmqType);
+        obj.pushKV("version", int{nVersion});
```
[the remark this answers is not in the document]
Comment (PastaPastaPasta) [reply src/llmq/commitment.h:103]: Why? Using braces gives us compile time guarantee there's no loss

[the remark this answers is not in the document]
Comment (UdjinM6) [reply src/llmq/commitment.h:103]: Ah.. makes sense. Shouldn't we use `{}` for `nVersion` in `CFinalCommitmentTxPayload::ToJson()` below too then?

[the remark this answers is not in the document]
Comment (PastaPastaPasta) [reply src/llmq/commitment.h:103]: Yeah, it should

**Comment by UdjinM6** [comment]
Looks good but linter complains now 🙈 
```
src/llmq/chainlocks.cpp:30:5: warning: Variable 'scheduler_thread' is assigned in constructor body. Consider performing initialization in initialization list. [useInitializationList]
```
so we need either
```diff
diff --git a/src/llmq/chainlocks.cpp b/src/llmq/chainlocks.cpp
index 6d60200b9a..54fa4b0024 100644
--- a/src/llmq/chainlocks.cpp
+++ b/src/llmq/chainlocks.cpp
@@ -25,9 +25,9 @@ CChainLocksHandler* chainLocksHandler;
 
 CChainLocksHandler::CChainLocksHandler(CTxMemPool& _mempool, CConnman& _connman) :
     scheduler(std::make_unique<CScheduler>()),
-    mempool(_mempool), connman(_connman)
+    mempool(_mempool), connman(_connman),
+    scheduler_thread(std::make_unique<std::thread>([&] { TraceThread("cl-schdlr", [&] { scheduler->serviceQueue(); }); }))
 {
-    scheduler_thread = std::make_unique<std::thread>([&] { TraceThread("cl-schdlr", [&] { scheduler->serviceQueue(); }); });
 }
 
 CChainLocksHandler::~CChainLocksHandler()
```
or add it to the ignore list

**Review by UdjinM6 — APPROVED** [review]
LGTM, utACK

**Pull request merged**